### PR TITLE
Create specs's page database

### PR DESCRIPTION
**Pull request by MatShir**
The first bases of specs

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Specs of the database page

### CODE REVIEW

**Comment by MatShir** [comment]
Simple specs the new coming fix @PierreRambaud @Robin-Fischer-PS, don't hesitate to review it and compare it also with the code if there is anything missing :hugs:

File: back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md [hunk 1]
```diff
@@ -1,8 +1,59 @@
...
+The feature to hide the password or the secure key will be removed since we can't fully hide them since with an alias the keys or passwords can be displayed.
 
-Row action does not apply to the checkbox column.
+## 1.7.7.0 Version
+
+This tool enables you to perform SQL queries directly on the PrestaShop database, and save them for use at any later time. 
+
+For instance, using this tool and your knowledge of SQL, you could create a reusable query giving you an updated list of the clients who are subscribed to your newsletter or get a list of products in HTML or CSV format.
+
+For security reasons, some types of queries are not allowed: UPDATE, DELETE, CREATE TABLE, DROP, etc. In short, you can only read data (SELECT query).
```
Comment (PierreRambaud) [inline back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md:15]: Unauthorized keywords:
        'unauthorized' => [
            'DELETE', 'ALTER', 'INSERT', 'REPLACE', 'CREATE', 'TRUNCATE', 'OPTIMIZE', 'GRANT', 'REVOKE', 'SHOW', 'HANDLER',
            'LOAD', 'ROLLBACK', 'SAVEPOINT', 'UNLOCK', 'INSTALL', 'UNINSTALL', 'ANALZYE', 'BACKUP', 'CHECK', 'CHECKSUM', 'REPAIR', 'RESTORE', 'CACHE',
            'DESCRIBE', 'EXPLAIN', 'USE', 'HELP', 'SET', 'DUPLICATE', 'VALUES',  'INTO', 'RENAME', 'CALL', 'PROCEDURE',  'FUNCTION', 'DATABASE', 'SERVER',
            'LOGFILE', 'DEFINER', 'RETURNS', 'EVENT', 'TABLESPACE', 'VIEW', 'TRIGGER', 'DATA', 'DO', 'PASSWORD', 'USER', 'PLUGIN', 'FLUSH', 'KILL',
            'RESET', 'START', 'STOP', 'PURGE', 'EXECUTE', 'PREPARE', 'DEALLOCATE', 'LOCK', 'USING', 'DROP', 'FOR', 'UPDATE', 'BEGIN', 'BY', 'ALL', 'SHARE',
            'MODE', 'TO', 'KEY', 'DISTINCTROW', 'DISTINCT',  'HIGH_PRIORITY', 'LOW_PRIORITY', 'DELAYED', 'IGNORE', 'FORCE', 'STRAIGHT_JOIN',
            'SQL_SMALL_RESULT', 'SQL_BIG_RESULT', 'QUICK', 'SQL_BUFFER_RESULT', 'SQL_CACHE', 'SQL_NO_CACHE', 'SQL_CALC_FOUND_ROWS', 'WITH',
        ],

File: back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md [hunk 2]
```diff
@@ -1,8 +1,59 @@
 # SPECIFICATIONS - ADVANCED PARAMETERS &gt; DATABASE &gt; SQL MANAGER
 
-## Grid
+The SQL manager is a complex feature, which should be reserved for technical people who know how to explore a database using the SQL language. In return for this complexity, it can be extremely powerful and proves immensely useful to those who master it.
 
-Default action in the button: Export Other available actions in the menu: View, Edit, Delete Row action: View
+## 1.7.8 Version 
+The page is going to be migrated into a module. The module will be downloadable for merchants who need it.
```
Comment (marionf) [inline back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md:6]: Native module ? Not pre-installed ?

File: back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md [hunk 3]
```diff
@@ -1,8 +1,59 @@
...
+    - You can also export the query results as a CSV file by clicking on the Export button
+"
+
+- Alert warning:
+
+"When saving the query, only the "SELECT" SQL statement is allowed."
+
+### SQL Manager List
+
+Below the messages, the page displays the SQL Manager list with the request's ID, SQL query name, and action columns.
```
Comment (marionf) [inline back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md:33]:
```suggestion
Below the messages, the page displays the SQL Manager list with the request's ID, SQL query name, SQL query and actions columns.
```
```suggestion
Below the messages, the page displays the SQL Manager list with the request's ID, SQL query name, and action columns.
```

File: back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md (hunk 2 above)
Comment (MatShir) [reply back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md:6]: It is not defined for now but it seems to be not pre-installed

File: back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md [hunk 4]
```diff
@@ -1,8 +1,59 @@
...
+The request's actions allow us to download the request's result into a CSV file, view the request's result on the page, delete the request, and edit it.
+To summarize the actions cab be:
+- Export. Performs the query, and have it downloaded by you in CSV format.
```
Comment (marionf) [inline back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md:36]: What do you mean by "performs the query" ? 
The query is just downloaded in CSV format

File: back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md [hunk 5]
```diff
@@ -1,8 +1,59 @@
 # SPECIFICATIONS - ADVANCED PARAMETERS &gt; DATABASE &gt; SQL MANAGER
 
-## Grid
+The SQL manager is a complex feature, which should be reserved for technical people who know how to explore a database using the SQL language. In return for this complexity, it can be extremely powerful and proves immensely useful to those who master it.
 
-Default action in the button: Export Other available actions in the menu: View, Edit, Delete Row action: View
+## 1.7.8 Version 
+The page is going to be migrated into a module. The module will be downloadable for merchants who need it.
+The feature to hide the password or the secure key will be removed since we can't fully hide them since with an alias the keys or passwords can be displayed.
```
Comment (marionf) [inline back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md:7]: Are you talking of that ?

capture d'écran_2829

File: back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md (hunk 4 above)
Comment (MatShir) [reply back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md:38]: Execute the query to retrieve the data and then download it. It is not the query that is exported but the results of the query. I have doubts now, I'm going to check

Comment (marionf) [reply back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md:38]: Ah, yes my bad you are right it's the results of the query !
So the query is well executed to get the results :)

File: back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md [hunk 6]
```diff
@@ -32,7 +32,7 @@ At first, the SQL Manager page displays 2 messages:
 
 ### SQL Manager List
 
-Below the messages, the page displays the SQL Manager list with the request's ID, SQL query name, and action columns.
+Below the messages, the page displays the SQL Manager list with the request's ID, SQL query name, SQL query and actions columns.
 The request's actions allow us to download the request's result into a CSV file, view the request's result on the page, delete the request, and edit it.
 To summarize the actions cab be:
 - Export. Performs the query, and have it downloaded by you in CSV format.
```
Comment (marionf) [inline back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md:38]:
```suggestion
- Export. Performs the query, and have the query's results downloaded in CSV format.
```

**Review by marionf — CHANGES_REQUESTED** [review]

File: back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md [hunk 7]
```diff
@@ -1,8 +1,70 @@
...
+
+"When saving the query, only the "SELECT" SQL statement is allowed."
+
+### SQL Manager List
+
+Below the messages, the page displays the SQL Manager list with the request's ID, SQL query name, SQL query and actions columns.
+The request's actions allow us to download the request's result into a CSV file, view the request's result on the page, delete the request, and edit it.
+To summarize the actions cab be:
+- Export. Performs the query, and have it downloaded by you in CSV format.
+- View. Performs the query, and has it displayed in an HTML table, right within the PrestaShop interface.
```
Comment (marionf) [inline back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md:39]:
```suggestion
- View. Performs the query, and get the results displayed in an HTML table, right within the PrestaShop interface.
```

File: back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md [hunk 8]
```diff
@@ -1,8 +1,70 @@
...
+
+Creating a new query is divided into 3 sections: 
+
+- The SQL form with 2 following inputs : 
+
+    - SQL query name. Make the name as long and descriptive as necessary.
+    - SQL query. The SQL query itself. You are free to perform JOINs or other intricate selections.
+ The cancel button redirects the user to the SQL Manager list without saving any information.
+ 
+ The save button saves the SQL name and request and redirects the merchant to the list with the new request saved. If the merchant saves a request that has not the authorized querie type (defined in the list bellow), a error is displayed : "Your request is not authorised, please try a SELECT type request", the query is not saved and the user stays on the page with the displayed error.
```
Comment (marionf) [inline back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md:58]:
```suggestion
 The save button saves the SQL query name, the SQL query and redirects the user to the list with the new SQL query saved. If the user saves a request that has not the authorized query type (defined in the list bellow), an error is displayed : "Your request is not authorised, please try a SELECT type request", the query is not saved and the user stays on the page with the error displayed.
```

**Review by marionf — CHANGES_REQUESTED** [review]

File: back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md [hunk 9]
```diff
@@ -1,8 +1,70 @@
...
 
-## Grid
+The SQL manager is a complex feature, which should be reserved for technical people who know how to explore a database using the SQL language. In return for this complexity, it can be extremely powerful and proves immensely useful to those who master it.
 
-Default action in the button: Export Other available actions in the menu: View, Edit, Delete Row action: View
+## 1.7.8 Version 
 
-Row action does not apply to the checkbox column.
+The page is going to be migrated into a module. The type of module is still to be defined, the most likely solution seems to be an not installed module.
+The module will be downloadable for merchants who need it and have the required skills.
```
Comment (marionf) [inline back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md:8]:
```suggestion
The module will be downloadable for users who need it and have the required skills.
```

**Review by marionf — CHANGES_REQUESTED** [review]

**Comment by MatShir** [comment]
Add the succes alert when saving :notebook:

**Review by marionf — APPROVED** [review]

File: back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md (hunk 5 above)
Comment (MatShir) [reply back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md:9]: Yes, @PierreRambaud was saying that it can be easily bypassed.

**Comment by MatShir** [comment]
Hi @Julievrz, 
Could check the wording for the error when the SQL is not saved, please? It is a new wording that is going to be integrated on the 178.

File: back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md (hunk 5 above)
Comment (marionf) [reply back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md:9]: All right, while reading I thougt you were talking of a feature like a "Show / hide password" button :smile: 
Maybe it will be more clearer if you add a screenshot ? Or say directly that they will be displayed in clear?

**Comment by MatShir** [comment]
:warning: Will be merged after the QA and the Product Content approved the review

File: back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md [hunk 10]
```diff
@@ -1,8 +1,74 @@
...
+
+### Settings
+
+Select your default file encoding. You can configure the character encoding of the downloaded CSV file. The default, UTF-8, is recommended, but you can select ISO-8859-1 if need be.
+
+### Add new SQL query
+
+Creating a new query is divided into 3 sections: 
+
+- The SQL form with 2 following inputs :
```
Comment (LouiseBonnard) [inline back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md:53]:
```suggestion
- The SQL form has 2 following input fields:
```

File: back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md [hunk 11]
```diff
@@ -1,8 +1,74 @@
...
+
+Select your default file encoding. You can configure the character encoding of the downloaded CSV file. The default, UTF-8, is recommended, but you can select ISO-8859-1 if need be.
+
+### Add new SQL query
+
+Creating a new query is divided into 3 sections: 
+
+- The SQL form with 2 following inputs : 
+
+    - SQL query name. Make the name as long and descriptive as necessary.
```
Comment (LouiseBonnard) [inline back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md:55]:
```suggestion
    - SQL query name. Users can type whatever name they need, there is no limitation in the number or the characters allowed.
```

File: back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md [hunk 12]
```diff
@@ -1,8 +1,74 @@
...
+    - SQL query. The SQL query itself. You are free to perform JOINs or other intricate selections.
```
Comment (LouiseBonnard) [inline back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md:56]: What do we expect here?

File: back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md [hunk 13]
```diff
@@ -1,8 +1,74 @@
...
+
+The cancel button redirects the user to the SQL Manager list without saving any information.
+ 
+The save button saves the SQL query name, the SQL query, and redirects the user to the list adding the new SQL query saved with the success alert: "Successful creation.".
```
Comment (LouiseBonnard) [inline back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md:60]:
```suggestion
The save button saves the SQL query name, the SQL query, and redirects the user to the list adding the new SQL query saved with the success notification: "Successful creation.".
```

File: back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md [hunk 14]
```diff
@@ -1,8 +1,74 @@
...
+'DESCRIBE', 'EXPLAIN', 'USE', 'HELP', 'SET', 'DUPLICATE', 'VALUES', 'INTO', 'RENAME', 'CALL', 'PROCEDURE', 'FUNCTION', 'DATABASE', 'SERVER',
+'LOGFILE', 'DEFINER', 'RETURNS', 'EVENT', 'TABLESPACE', 'VIEW', 'TRIGGER', 'DATA', 'DO', 'PASSWORD', 'USER', 'PLUGIN', 'FLUSH', 'KILL',
+'RESET', 'START', 'STOP', 'PURGE', 'EXECUTE', 'PREPARE', 'DEALLOCATE', 'LOCK', 'USING', 'DROP', 'FOR', 'UPDATE', 'BEGIN', 'BY', 'ALL', 'SHARE',
+'MODE', 'TO', 'KEY', 'DISTINCTROW', 'DISTINCT', 'HIGH_PRIORITY', 'LOW_PRIORITY', 'DELAYED', 'IGNORE', 'FORCE', 'STRAIGHT_JOIN',
+'SQL_SMALL_RESULT', 'SQL_BIG_RESULT', 'QUICK', 'SQL_BUFFER_RESULT', 'SQL_CACHE', 'SQL_NO_CACHE', 'SQL_CALC_FOUND_ROWS', 'WITH'.
+Each keywords are checked in the SQL query before saving it.
+
+On the bottom left, the "List of MySQL Tables" gives the user a list of all the available database tables. Select a table to make PrestaShop display its attributes and types on the list of attributes for this MySQL table, then click "Add attribute to SQL query" or "Add table name to SQL query" to send its attribute name or table name into the SQL query field.
```
Comment (LouiseBonnard) [inline back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md:74]:
```suggestion
On the bottom left, the "List of MySQL Tables" gives the user a list of all the available database tables. He/she should select a table to make PrestaShop display all its attributes and types in the list on the right, then click "Add attribute to SQL query" or "Add table name to SQL query" to send its attribute name/table name into the SQL query field.
```

File: back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md [hunk 15]
```diff
@@ -1,8 +1,74 @@
...
+Creating a new query is divided into 3 sections: 
+
+- The SQL form with 2 following inputs : 
+
+    - SQL query name. Make the name as long and descriptive as necessary.
+    - SQL query. The SQL query itself. You are free to perform JOINs or other intricate selections.
+
+The cancel button redirects the user to the SQL Manager list without saving any information.
+ 
+The save button saves the SQL query name, the SQL query, and redirects the user to the list adding the new SQL query saved with the success alert: "Successful creation.".
```
Comment (Julievrz) [inline back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md:60]:
```suggestion
The save button saves the SQL query name, the SQL query, and redirects the user to the list adding the new SQL query saved with the success alert: "SQL query was successfully saved."
```

File: back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md [hunk 16]
```diff
@@ -1,8 +1,74 @@
...
+
+If the user saves a request that has not the authorized query type (defined in the list below), an error is displayed: "Your request is not authorized, please try a SELECT type request", the query is not saved and the user stays on the page with the error displayed.
```
Comment (Julievrz) [inline back-office/advanced-parameters/database/advanced_parameters-database-sql_manager.md:62]:
```suggestion
If the user saves a request that has not the authorized query type (defined in the list below), an error is displayed: "Error: only SELECT type queries are authorized.", the query is not saved and the user stays on the page with the error displayed.
```

**Review by LouiseBonnard — APPROVED** [review]

**Comment by marionf** [comment]
Thank you @MatShir !
